internal/wasm/engine.go

```diff
@@ -52,3 +53,13 @@ type ModuleEngine interface {
 	// the initialization via ElementSegment.
 	FunctionInstanceReference(funcIndex Index) Reference
 }
+
+// EngineConfig caries extra configuration for an Engine.
+type EngineConfig struct {
+	// EnabledPerfmap enable the generation of perfmap files used for profiling.
+	EnabledPerfmap bool
+
+	EnabledFeatures api.CoreFeatures
+
+	FileCache filecache.Cache
```

Comment:
these need comments!
```suggestion
	// EnabledFeatures
	EnabledFeatures api.CoreFeatures
	// FileCache
	FileCache filecache.Cache
```